runtime/vm/LayoutFFITypeHelpers.cpp

```diff
@@ -103,25 +104,88 @@ LayoutFFITypeHelpers::getArrayFFIType(char **layout, UDATA nElements) {
 ffi_type*
 LayoutFFITypeHelpers::getStructFFIType(char **layout)
 {
+	char *currentLayout = *layout;
+	J9LayoutStrFFITypeEntry *layoutStrFFITypeEntry = NULL;
+	J9LayoutStrFFITypeEntry *resultEntry = NULL;
+	ffi_type **structElements = NULL;
 	ffi_type *typeFFI = NULL;
+	PORT_ACCESS_FROM_JAVAVM(_vm);
+
+	if (NULL == _vm->layoutStrFFITypeTable) {
+		_vm->layoutStrFFITypeTable = createLayouStrFFITypeTable(_vm);
+		if (NULL == _vm->layoutStrFFITypeTable) {
+			goto freeAllMemoryThenExit;
+		}
+	}
+
+	layoutStrFFITypeEntry = (J9LayoutStrFFITypeEntry *)j9mem_allocate_memory(sizeof(J9LayoutStrFFITypeEntry), J9MEM_CATEGORY_VM_FFI);
+	if (NULL == layoutStrFFITypeEntry) {
+		goto freeAllMemoryThenExit;
+	}
+
+	/* Search the hashtable for the ffi_type if the same strcut layout string exists. */
+	currentLayout += 1; /* Skip over the '#' case for the length of the struct layout string. */
```

Comment:
I think an assertion is warranted here rather than just assume the character is `#`.

Reply:
Not really as  the symbol `#` is identified by all the code calling `LayoutFFITypeHelpers::getStructFFIType()` e.g.
https://github.com/eclipse-openj9/openj9/blob/d57675ee3710cbc3e2fa54ebbe875956f0ffb924/runtime/vm/LayoutFFITypeHelpers.cpp#L149
https://github.com/eclipse-openj9/openj9/blob/d57675ee3710cbc3e2fa54ebbe875956f0ffb924/runtime/vm/LayoutFFITypeHelpers.cpp#L68